### DISCUSSION TITLE
Connect isolated clusters

### DISCUSSION THREAD

**Discussion by Sgobro** [PMEAL/OpenPNM#2747] [Q&A]
Hi,

I connected all the isolated pores and now I am trying to connected the isolated clusters of pores but I don't know how to do that. Both the function "stitch" and "merge" require as input two networks, instead I just have the array of the disconnected pores that I want to connect to the bigger cluster on my network. 

Thanks

**Comment by jgostick** [comment]
Since what you're doing is quite specific there is not a pre-written function for it.  However you can use the `op.topotools.extend` function to just add new throats to the network by hand.  The tricky part is to decide which pores on the cluster(s) should connect to which pores on the network.  Actually an option that occurs to me is that you could actually create a second network that is clone of the first, then delete the clusters from network A and delete everything BUT the clusters on network B, then you could use the stitch network function?

**Reply by Sgobro** [reply]
Thanks for your reply, now I will try your way. For now I connected disconnected pores to the nearest pore belonging to the biggest cluster in my porenetwork; but I am afraid it connects disconnected pores to pores too much far from each other.

This is my code:
<img width="424" alt="github" src="https://github.com/PMEAL/OpenPNM/assets/129179101/06a65b70-8ae4-4ab3-a6d6-3ea8a859e337">

**Reply by Sgobro** [reply]
Thanks a lot, I tried as you suggested to create these two network and to stitch them, I obtained a fully connected network.

Again thanks for your time.

**Reply by Sgobro** [reply]
@jgostick Just another question. I plotted the data sheet of the throats of my network and I noticed that I obtained NaN values for all the features of the new generated throats (like throat.length, throat.diameter, etc..), I also regenerated the model but still the throat.length and total_length of the new throats are NaN, how can I fix it? Should I calculate each lenght one by one and assign them to the network?
Thanks a lot.
Cesare

**Reply by jgostick** [reply]
You can actually 'mix and match' model now.  So you can assign 'throat.length' to the "throat.all" domain.  This way you can save a lot of effort by only duplicating the models which are different between the domains.  Hopefully that helps.

**Comment by Sgobro** [comment]
Hi Professor ,I think I didn't understand what do you mean.

**Reply by jgostick** [reply]
You can have some models assigned the 'pore.all', and others assigned to 'pore.left' or 'pore.right'.  In the old way (openpnm v2) you need to assign a whole set of models to the geometry of the 'left' and a whole second set assigned to 'right', even if most of the models were the same between the two.

**Reply by Sgobro** [reply]
Thanks again for the reply Professor. I am using the last version of OpenPNM so how to do that in the last one?
For now I assigned to the new throats the distance between the pores connected through that throat as throat.direct_length and I noticed a slightly difference in the absolute permeability value but still there is the throat.total_lenght which has NaN values assigned to the new connections.

**Reply by jgostick** [reply]
I'm sorry, I have kinda lost track of your problem.  Even re-reading this thread doesn't help.  Can you try to reformulate your current problem and ask it as a new Discussion?
